### PR TITLE
chore: update faraday dependency range

### PR DESCRIPTION
**Pull request by daedric**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no     
| Need Doc update   | no

## Describe your change
Update Faraday dependency.

## What problem is this fixing?
Strong dependency on "old" Faraday major.

### CODE REVIEW

**Comment by daedric** [comment]
I was not able to run the test however in some private code I was able to use Faraday 2.2 without any issue :)

**Comment by sfate** [comment]
@DevinCodes any news on this one?

File: algolia.gemspec [hunk 1]
```diff
@@ -35,7 +35,8 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake'
   spec.add_development_dependency 'rubocop', '<= 0.82.0'
 
-  spec.add_dependency 'faraday', ['>= 0.15', '< 2.0']
+  spec.add_dependency 'faraday', ['>= 2', '< 3']
```
Comment (DevinCodes) [inline algolia.gemspec:38]: We may want to allow older versions here too. If someone has the faraday dependency in their project this range may prevent them from upgrading. WDYT?
```suggestion
  spec.add_dependency 'faraday', ['>= 0.15', '< 3']
```

File: algolia.gemspec [hunk 2]
```diff
@@ -35,7 +35,8 @@ Gem::Specification.new do |spec|
...
+  spec.add_dependency 'faraday-net_http_persistent', '~> 2.0'
```
Comment (DevinCodes) [inline algolia.gemspec:39]: If we go that route, we should probably have the same range for this dependency
```suggestion
  spec.add_dependency 'faraday-net_http_persistent', ['>= 0.15', '< 3']
```

**Review by DevinCodes — CHANGES_REQUESTED** [review]
Thank you for the PR! I have a remark, let me know what you think 🙂

**Comment by aaronstillwell** [comment]
@DevinCodes @daedric any help needed here to get this merged? I'm unfortunately blocked from using `algoliasearch-rails` until this is resolved - happy to contribute!

File: algolia.gemspec (hunk 1 above)
Comment (daedric) [reply algolia.gemspec:38]: @DevinCodes Faraday 2 is not backward compatible, I cannot use this range (IIRC)

**Comment by daedric** [comment]
Done, I'm sorry I missed the (old) notification :)

**Comment by DevinCodes** [comment]
Quick update; this change seems to require us to drop support for some older Ruby versions, so we're internally discussing how we can address this properly. Will keep you posted!

**Comment by courtsimas** [comment]
👀 

this ruby gem prevents the gem `google-cloud-pubsub`  from resolving due to their `~> 2.0` on faraday. Any update?

**Comment by aaronstillwell** [comment]
While we wait for the team to get back to us on their decision moving forward, we can work against @daedric's fork with:

```ruby
gem 'algolia', git: 'https://github.com/daedric/algoliasearch-client-ruby', ref: '911b338'
```

**Comment by bvogel** [comment]
> Quick update; this change seems to require us to drop support for some older Ruby versions, so we're internally discussing how we can address this properly. Will keep you posted!

I would really move froward and stop supporting ruby versions that are well behind EOL, even 2.6 is now out of support so even maintaining 2.5 would be generous. As for faraday which you rely upon they don't support ruby versions after their EOL and even the support for 0.x was stopped as only the 1.x and 2.x received releases lately. 

I don't understand why algolia is so desperately lagging in version updates as this general approach of "leave no one behind" will not foster a more general consensus for developers to keep upgrading their dependencies and by that keeping their applications unsafe as even security fixes aren't applied bc of "version constraints"

**Comment by DevinCodes** [comment]
Sorry for the delay in coming back to you. We're indeed going to drop support for the older Ruby versions. We're not going to release today given that it's a Friday, but I'm working on this first thing Monday morning (Paris time). I'll drop a note here when a release is made!

**Review by DevinCodes — APPROVED** [review]

**Pull request merged**